week1/plot_trips.R

```diff
@@ -1,62 +1,186 @@
-
-# repeat this, splitting results by whether there was substantial precipitation or not
-# you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
-
-# add a smoothed fit on top of the previous plot, using geom_smooth
-
-# compute the average number of trips and standard deviation in number of trips by hour of the day
-# hint: use the hour() function from the lubridate package
-
-# plot the above
-
-# repeat this, but now split the results by day of the week (Monday, Tuesday, ...) or weekday vs. weekend days
-# hint: use the wday() function from the lubridate package
+########################################
+# load libraries
+########################################
+
+# load some packages that we'll need
+library(tidyverse)
+library(scales)
+library(lubridate)
+library(cowplot)
+
+# be picky about white backgrounds on our plots
+theme_set(theme_bw())
+
+# load RData file output by load_trips.R
+load('trips.RData')
+########################################
+
+
+########################################
+# plot trip data
+
+# plot the distribution of trip times across all rides (compare a histogram vs. a density plot)
+summary(trips$tripduration)
+
+hist1 <- trips %>%
+  ggplot(aes(x = tripduration/60)) +
+  geom_histogram(bins = 100, color='red', fill = 'red', alpha = 0.4) + 
+  scale_x_continuous(limits = c(0, 120), breaks = pretty_breaks(n=6)) +
+  scale_y_continuous(label = comma, breaks = pretty_breaks(n=8)) +
+  labs(x = "Trip duration in minutes", y="no. of trips", title = "Trip times across all rides")
+
+dense1 <- trips %>%
+  ggplot(aes(x = tripduration/60)) +
+  geom_density(color= 'blue', fill = 'blue', alpha = 0.4) + 
+  scale_x_continuous(limits = c(0, 120), breaks = pretty_breaks(n=12)) +
+  scale_y_continuous(label = comma, breaks = pretty_breaks(n=8)) +
+  labs(x = "Trip duration in minutes", title = "Trip times across all rides")
+
+plot_grid(hist1, dense1, ncol = 1)
+
+
+
+# plot the distribution of trip times by rider type indicated using color and fill (compare a histogram vs. a density plot)
+hist2 <- trips %>%
+  ggplot(aes(x = tripduration/60, fill = usertype, color = usertype, alpha = 0.7)) +
+  geom_histogram(bins=60) +
+  xlim(0, 60) +
+  scale_y_continuous(label=comma) +
+  facet_wrap(~ usertype, scale = 'free_y') +
+  labs(x = "Trip duration in minutes", title = "Trip times across all rides")
+
+dense2 <- trips %>%
+  ggplot(aes(x = tripduration/60, fill = usertype)) +
+  geom_density() +
+  xlim(0, 60) +
+  facet_wrap(~ usertype) +
+  labs(x = "Trip duration in minutes", title = "Trip times across all rides")
+
+plot_grid(hist2, dense2, ncol = 1)
+
+
+# plot the total number of trips on each day in the dataset
+trips %>%
+  group_by(ymd) %>%
+  summarize(count = n()) %>%
+  ggplot(aes(x = ymd, y= count)) +
+  geom_point(aes(color = factor(month(ymd), levels= 1:12, labels = month.name))) + 

```

Comment:
What is the levels here doing? And where did you get 1:12 from?

Reply:
Ah I see - pulling the month. Why do that in aes vs. doing it in a mutate function? vs separating it into a new line could make it (1) more extensible and (2) readable